File: 121.best-time-to-buy-and-sell-stock.go
package leetcode

// @leet start
func maxProfit(prices []int) int {
	curMin, best := prices[0], 0
	for i := 1; i < len(prices); i++ {
		if prices[i] < curMin {
			curMin = prices[i]
		}
		best = max(best, prices[i]-curMin)
	}
	return best
}

// @leet end

File: TwoPointer/121.best-time-to-buy-and-sell-stock.py
from typing import List


# @leet start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res, low = 0, prices[0]
        for v in prices:
            low = min(v, low)
            res = max(v - low, res)

        return res


# @leet end

